memory_set/src/area.rs

```diff
@@ -113,28 +98,37 @@ impl<F: Copy, P, B: MappingBackend<F, P>> MemoryArea<F, P, B> {
     ///
     /// The start address of the memory area is increased by `new_size`. The
     /// shrunk part is unmapped.
-    pub(crate) fn shrink_left(&mut self, new_size: usize, page_table: &mut P) -> MappingResult {
+    pub(crate) fn shrink_left(
+        &mut self,
+        new_size: usize,
+        page_table: &mut B::PageTable,
+    ) -> MappingResult {
         let unmap_size = self.size() - new_size;
         if !self.backend.unmap(self.start(), unmap_size, page_table) {
             return Err(MappingError::BadState);
         }
-        self.va_range.start += unmap_size;
+        self.va_range.start = (self.va_range.start.into() + unmap_size).into();
```

Comment:
Why `AddAssign` does not work?

Reply:
Because `MemoryAddr` is `Copy + From<usize> + Into<usize>`, not `AddAssign<usize>`

Reply:
> Because `MemoryAddr` is `Copy + From<usize> + Into<usize>`, not `AddAssign<usize>`

I must say that I still think it's worth it to make `MemoryAddr` also `Add<usize> + AddAssign<usize> + Sub<usize> + SubAssign<usize> + Sub<Output = usize>`, maybe also `Ord`. Converting them from/to usize is annoying.